contracts/payloads/IGP26/PayloadIGP26.sol

```diff
@@ -0,0 +1,662 @@
+            return 0xc5911Fa3917c507fBEbAb910C8b47cBdD3Ce147e; // VAULT_WSTETH_USDC
+        } else if (vaultId == 15) {
+            return 0x38aE6fa3d6376D86D1EE591364CD4b45C99adE22; // VAULT_WSTETH_USDT
+        } else if (vaultId == 16) {
+            return 0xEA0C58bE3133Cb7f035faCF45cb1d4F84CF178B4; // VAULT_WEETH_WSTETH
+        } else if (vaultId == 17) {
+            return 0x72DB9B7Bd2b0BC282708E85E16123023b32de6A9; // VAULT_SUSDE_USDC
+        } else if (vaultId == 18) {
+            return 0x72DB9B7Bd2b0BC282708E85E16123023b32de6A9; // VAULT_SUSDE_USDT
+        } else if (vaultId == 19) {
+            return 0xda8a70b9533DEBE425F8A3b2B33bc09c0415e5FE; // VAULT_WEETH_USDC
+        } else if (vaultId == 20) {
+            return 0x32eE0cB3587C6e9f8Ad2a0CF83B6Cf326848b7c6; // VAULT_WEETH_USDT
+        } else {
+            revert ("no oracle address");
+        }
+    }
+
+    function getVaultConfig(address vault) internal view returns (IFluidVaultT1.Configs memory configs) {
+        uint vaultVariables2 = IFluidVaultT1(vault).readFromStorage(bytes32(uint256(1)));
+        configs.supplyRateMagnifier = uint16(vaultVariables2 & X16);
+        configs.borrowRateMagnifier = uint16((vaultVariables2 >> 16) & X16);
+        configs.collateralFactor = (uint16((vaultVariables2 >> 32) & X10)) * 10;
+        configs.liquidationThreshold = (uint16((vaultVariables2 >> 42) & X10)) * 10;
+        configs.liquidationMaxLimit = (uint16((vaultVariables2 >> 52) & X10) * 10);
+        configs.withdrawalGap = uint16((vaultVariables2 >> 62) & X10) * 10;
+        configs.liquidationPenalty = uint16((vaultVariables2 >> 72) & X10);
+        configs.borrowFee = uint16((vaultVariables2 >> 82) & X10);
+        configs.oracle = address(uint160(vaultVariables2 >> 96));
+    }
+
+    function cloneVault(uint256 oldVaultId) internal {
+        address oldVaultAddress = VAULT_T1_FACTORY.getVaultAddress(oldVaultId);
+        address newVaultAddress = VAULT_T1_FACTORY.getVaultAddress(oldVaultId + 10);
+
+        IFluidVaultT1.ConstantViews memory oldConstants = IFluidVaultT1(oldVaultAddress).constantsView();
+        IFluidVaultT1.ConstantViews memory newConstants = IFluidVaultT1(newVaultAddress).constantsView();
+
+        address newOracleAddress = getOracleAddress(oldVaultId + 10);
+
+        {
+            require(oldConstants.supplyToken == newConstants.supplyToken, "not-same-supply-token");
+            require(oldConstants.borrowToken == newConstants.borrowToken, "not-same-borrow-token");
+        }
+        
+        // Set user supply config for the vault on Liquidity Layer.
+        {
+            AdminModuleStructs.UserSupplyConfig[]
+                memory configs_ = new AdminModuleStructs.UserSupplyConfig[](1);
+
+            configs_[0] = getUserSupplyData(newConstants.supplyToken, oldVaultAddress);
+
+            (uint256 baseAllowance, ) = getAllowance(newConstants.supplyToken);
+
+            configs_[0].baseWithdrawalLimit = baseAllowance;
```

Comment:
we have a terrible UX at Liquidity AdminModule so base withdrawal limit input must be in raw, same for base debt ceiling and max debt ceiling :smiling_face_with_tear: I.e. currently the input is in normal token amount, but it must be `normalTokenAmount * 1e12 / exchangePrice`. 

we can get the exchange price for a token like this: 
```
uint256 exchangePriceAndConfig_ = LIQUIDITY.readFromStorage(
                LiquiditySlotsLink.calculateMappingStorageSlot(
                    LiquiditySlotsLink.LIQUIDITY_EXCHANGE_PRICES_MAPPING_SLOT,
                    token_
                )
            );
(uint256 supplyExchangePrice, uint256 borrowExchangePrice) = LiquidityCalcs
                .calcExchangePrices(exchangePriceAndConfig_);
```

Reply:
@bergben Before we use to set normal amount, right?

Reply:
Resolved. Github is not showing me a resolve button here idk why